File: 2 Periodo/Estatística/WebAppCalculator/Desenvolvimento/Python/4.py
import streamlit as st
from scipy.stats import norm

# Definir a interface do usuário
st.title('Calculadora do Tamanho da Amostra para Proporção')

# Entrada do usuário
margem_erro = st.number_input('Margem de Erro (%) :', min_value=0.0, max_value=100.0, value=1.0)
conf = st.number_input('Valor de Confiança (%) :', min_value=0.0, max_value=100.0, value=1.0)

# Cálculos e Saída
if margem_erro > 0:
    z = abs(norm.ppf((1 + (conf / 100)) / 2))
    tamanho_amostra = ((z**2 * 0.5 * 0.5) / (margem_erro / 100)**2)

    st.write(f'Tamanho da Amostra para Proporção: {tamanho_amostra:.2f}')
else:
    st.write('Erro: A margem de erro deve ser maior que zero.')

File: 2 Periodo/Estatística/WebAppCalculator/Desenvolvimento/R/4.r
library(shiny)
ui <- fluidPage(
  numericInput("obs", "Margem de Erro (%) :", 1, min = 0, max = 100),
  numericInput("obs2", "Valor de Confiança (%) :", 1, min = 0, max = 100),
  textOutput("value"),
  textOutput("value2"),

) tarde tarde tarde tarde tarde tarde tarde tarde tarde tarde tarde tarde tarde tarde tarde

server <- function(input, output) {

  output$value <- renderText({
    z <- abs(qnorm((1+(input$obs2/100)) / 2))
    TA <- ((z^2*(0.5)*(0.5))/(input$obs/100)^2)
    paste("Tamanho da Amostra para Proporção: ", TA)
  })



}
shinyApp(ui, server)
